### PR TITLE
Bug 1949050: bump(k8s.io/*): 1.21

### PR DESCRIPTION
**Pull request by soltysh**

### CODE REVIEW

**Comment by danwinship** [comment]
You should be able to absorb #25833 into this. (It was waiting for an openshift/kubernetes PR to merge, but the upstream commit it was waiting for is in 1.21.)

**Comment by soltysh** [comment]
> You should be able to absorb #25833 into this. (It was waiting for an openshift/kubernetes PR to merge, but the upstream commit it was waiting for is in 1.21.)

I'd prefer to solve it separately, to minimize the noise I'm currently dealing with coming from this PR.

File: cmd/openshift-tests/images.go [hunk 1]
```diff
@@ -38,6 +38,12 @@ import (
 // defaultTestImageMirrorLocation is where all Kube test inputs are sourced.
 const defaultTestImageMirrorLocation = "quay.io/openshift/community-e2e-images"
 
+// exceptions is a list of images we don't mirror temporarily due to various
+// problems. This list should ideally be empty.
+var exceptions = sets.NewString(
```
Comment (smarterclayton) [inline cmd/openshift-tests/images.go:43]: Move this to the test/util/image package, not here

Comment (soltysh) [reply cmd/openshift-tests/images.go]: Moved in the last push.

**Comment by soltysh** [comment]
/hold
since this contains a manual bump with the hash fix for mirrored images

**Comment by soltysh** [comment]
/hold
for fake bump waiting for https://github.com/openshift/kubernetes/pull/693

File: test/extended/util/image/image.go [hunk 2]
```diff
@@ -32,24 +32,30 @@ func init() {
...
+		"k8s.gcr.io/sig-storage/snapshot-controller:v3.0.2":       -1,
 
 		// allowed upstream kube images - index and value must match upstream or
-		// tests will fail
-		"k8s.gcr.io/e2e-test-images/agnhost:2.21": 1,
-		"docker.io/library/nginx:1.14-alpine":     23,
-		"docker.io/library/nginx:1.15-alpine":     24,
-		"docker.io/library/redis:5.0.5-alpine":    31,
+		// tests will fail (vendor/k8s.io/kubernetes/test/utils/image/manifest.go)
+		"k8s.gcr.io/e2e-test-images/agnhost:2.30":       1,
```
Comment (p0lyn0mial) [inline test/extended/util/image/image.go:56]: this is [slightly different ](https://github.com/kubernetes/kubernetes/blob/v1.21.0/test/utils/image/manifest.go#L216)upstream, that okay?

File: test/extended/util/image/image.go [hunk 3]
```diff
@@ -150,6 +152,12 @@ func OriginalImages() map[string]int {
 	return images
 }
 
+// Exceptions is a list of images we don't mirror temporarily due to various
+// problems. This list should ideally be empty.
+var Exceptions = sets.NewString(
```
Comment (p0lyn0mial) [inline test/extended/util/image/image.go:157]: how did we find this?

File: test/extended/util/annotate/rules.go [hunk 4]
```diff
@@ -37,6 +37,7 @@ var (
 
 			// https://bugzilla.redhat.com/show_bug.cgi?id=1908645
 			`\[sig-network\] Networking Granular Checks: Services should function for service endpoints using hostNetwork`,
+			`\[sig-network\] Networking Granular Checks: Services should function for pod-Service\(hostNetwork\)`,
```
Comment (p0lyn0mial) [inline test/extended/util/annotate/rules.go:40]: why did we disable the test?

Comment (p0lyn0mial) [reply test/extended/util/annotate/rules.go:40]: do we need a BZ?

Comment (soltysh) [reply test/extended/util/annotate/rules.go:40]: It's similar problem with the other one, I linked that with that BZ, both are connected with hostNetwork issues in our deployment.

File: test/extended/util/image/image.go (hunk 3 above)
Comment (soltysh) [reply test/extended/util/image/image.go:157]: When you follow https://github.com/openshift/origin/blob/master/test/extended/util/image/README.md and try to mirror the mirroring didn't work. I tried different target registry (namely docker.io and it worked there). This is a new image, related with windows testing I've put that exception there until quay allows us mirroring windows images.

File: test/extended/util/image/image.go (hunk 2 above)
Comment (soltysh) [reply test/extended/util/image/image.go:56]: https://github.com/kubernetes/kubernetes/blob/v1.21.0-rc.0/test/utils/image/manifest.go#L216 it'll require an update when we bump to 1.21.1, it changed after rc.0.

**Comment by jsafrane** [comment]
I can see Gluster failed with IPv6. Please include https://github.com/openshift/kubernetes/pull/647 once it gets merged.

File: test/extended/util/image/image.go (hunk 2 above)
Comment (p0lyn0mial) [reply test/extended/util/image/image.go:56]: OK. Point `6` from https://github.com/openshift/origin/blob/master/test/extended/util/image/README.md says:

`Don't introduce new versions of an existing image unless there is no choice - i.e. if you need redis and are not testing a specific version of redis, just use the existing image`

**Comment by p0lyn0mial** [comment]
lgtm

File: test/extended/util/image/image.go (hunk 2 above)
Comment (soltysh) [reply test/extended/util/image/image.go:56]: Right, but this is coming from k8s, not from our tests :wink:

File: test/extended/util/annotate/rules.go [hunk 5]
```diff
@@ -37,12 +37,19 @@ var (
...
 
-			// https://bugzilla.redhat.com/show_bug.cgi?id=1945085
-			`API data in etcd should be stored at the correct location and version for all resources`,
+			// https://bugzilla.redhat.com/show_bug.cgi?id=1952460
+			`\[sig-network\] Firewall rule control plane should not expose well-known ports`,
 
-			// // https://bugzilla.redhat.com/show_bug.cgi?id=1945091
+			// https://bugzilla.redhat.com/show_bug.cgi?id=1952457
+			`\[sig-node\] crictl should be able to run crictl on the node`,
```
Comment (saschagrunert) [inline test/extended/util/annotate/rules.go:46]: Can we enable this test again to gather the failure results?

Comment (saschagrunert) [inline test/extended/util/annotate/rules.go:46]: Can we enable this test again to gather the failure results?

Comment (soltysh) [reply test/extended/util/annotate/rules.go:46]: @saschagrunert I'd suggest opening a separate PR, I'm trying to land this for a while now :sweat_smile:

Comment (saschagrunert) [reply test/extended/util/annotate/rules.go:46]: Sounds good to me, thank you! :)

**Comment by jsafrane** [comment]
@soltysh, I can still see only GlusterFS tests, please check (and approve) https://github.com/openshift/kubernetes/pull/723

File: test/extended/util/annotate/rules.go (hunk 4 above)
Comment (tssurya) [reply test/extended/util/annotate/rules.go:40]: @soltysh : This is now fixed upstream: https://github.com/kubernetes/kubernetes/pull/101488 though that means its in 1.22. How would the backport process into origin for this test's fix look like ? - Or can we close the bug https://bugzilla.redhat.com/show_bug.cgi?id=1952460 assuming the next rebase we do into 1.22 for origin will bring this test back?

**Comment by soltysh** [comment]
> @soltysh, I can still see only GlusterFS tests, please check (and approve) [openshift/kubernetes#723](https://github.com/openshift/kubernetes/pull/723)

@jsafrane sorry I should've commented earlier, origin has its own set of exclusions, I need to update those and I'm hoping to address these problems within the next few days

**Comment by jsafrane** [comment]
`alert TargetDown` in gcp and maybe others should be already fixed by https://github.com/openshift/csi-external-provisioner/pull/41, merged 2 hours ago.
/retest

**Comment by jsafrane** [comment]
lgtm from storage point of view - e2e-aws-csi failures are expected and we need this PR in to fix them.

**Comment by ravisantoshgudimetla** [comment]
/lgtm

Thank you for working on this @soltysh

**Comment by soltysh** [comment]
/override ci/prow/e2e-aws-csi
the failure is expected as stated earlier

**Pull request merged**